dask/base.py

```diff
@@ -15,12 +14,51 @@
 from .context import _globals
 from .utils import Dispatch, ignoring
 
-__all__ = ("Base", "compute", "normalize_token", "tokenize", "visualize")
+__all__ = ("Base", "is_dask_collection", "compute", "normalize_token",
+           "tokenize", "visualize")
 
 
-class Base(object):
+class DaskInterface(object):
+    """Minimal interface required for dask-like duck typing"""
+    def _dask_optimize_(self, dsk, keys, **kwargs):
+        """Return an optimized dask graph"""
+        raise NotImplementedError()
+
+    def _dask_finalize_(self, results):
+        """Finalize the computed keys into an output object"""
+        raise NotImplementedError()
+
+    def _dask_default_get_(self, dsk, key, **kwargs):
+        """The default scheduler `get` to use for this object"""
+        raise NotImplementedError()
```

Comment:
Why should `get` not be an instance method? Only thing I can think of is that it can't rely on `self`, because only one `get` is used. I did put a warning a couple places in the docs about it, so hopefully people just read about it there instead of mucking in the code here?